web/app/scripts/filterbar/filterbar-controller.js

```diff
@@ -54,9 +61,11 @@
         /**
          * When the record type changes, request the new schema
          */
-        $scope.$on('driver.state.recordstate:selected', function(event, selected) {
-            if (selected) {
+        $scope.$on('driver.state.recordstate:selected',
+                   function(event, selected) { onRecordSelected(selected); });
```

Comment:
onRecordSelected is wrapped so that the first argument isn't needed in the core function (which only uses `selected`)